system/system_monitor/src/cpu_monitor/cpu_monitor_base.cpp

```diff
@@ -198,18 +221,31 @@ void CPUMonitorBase::checkUsage(diagnostic_updater::DiagnosticStatusWrapper & st
           } else {
             whole_level = std::max(whole_level, level);
           }
+
+          if (cpu_name == "all") {
+            cpu_usage.all = cpu_status;
+          } else {
+            cpu_usage.cpus.push_back(cpu_status);
+          }
         }
       }
     }
   } catch (const std::exception & e) {
     stat.summary(DiagStatus::ERROR, "mpstat exception");
     stat.add("mpstat", e.what());
     std::fill(usage_check_cnt_.begin(), usage_check_cnt_.end(), 0);
+    cpu_usage.all.status = STATUS.STALE;
+    publishCpuUsage(cpu_usage);
+    cpu_usage.cpus.clear();
     return;
   }
 
   stat.summary(whole_level, load_dict_.at(whole_level));
 
+  // Publish msg
+  publishCpuUsage(cpu_usage);
+  cpu_usage.cpus.clear();
```

Comment:
Since it is automatically cleared when it go out of scope, clearing is not needed here.

Reply:
I removed it.